runner/file_test.go

```diff
@@ -570,32 +480,49 @@ c: d
 			inTmp, err := os.CreateTemp("", "")
 			assert.Nil(t, err)
 			defer inTmp.Close()
+			defer os.Remove(inTmp.Name())
 
 			inTmp.Write([]byte(test.in))
-
-			outTmp, err := os.CreateTemp("", "")
+			out, err := transformTestFile(inTmp.Name(), transformYAMLFile)
 			assert.Nil(t, err)
-			defer outTmp.Close()
 
-			jw, err := openJSONResultItemWriter(outTmp.Name(), nil)
-			assert.Nil(t, err)
-			rw := NewResultWriter(jw)
-			err = transformYAMLFile(inTmp.Name(), rw)
+			var exp any
+			err = jsonUnmarshal([]byte(test.exp), &exp)
 			assert.Nil(t, err)
 
-			rw.Close()
+			assert.Equal(t, out, exp)
+		})
+	}
+}
 
-			outTmpBs, err := ioutil.ReadFile(outTmp.Name())
-			assert.Nil(t, err)
+func transformTestFile[transformer func(string, *ResultWriter) error](filename string, transformFile transformer) (any, error) {
```

Comment:
Does this really need to be generic?

Reply:
It doesn't. Just started out that way :)

Reply:
Kk thanks!